### DISCUSSION TITLE
Some thoughts on the `node_injection` constraint

### DISCUSSION THREAD

**Discussion by mihlema** [spine-tools/SpineOpt.jl#861] [Ideas]
I am currently looking at our `node_injection` constraint and I believe there are a couple of potential bugs/need fixes:
The first comment relates to demand and fractional demands:
```
            + expr_sum(
                + node_injection[n, s, t] + demand[
                    (
                        node=n,
                        stochastic_scenario=s,
                        analysis_time=t0,
                        t=representative_time_slice(m, t),
                    ),
                ] for (n, s, t) in node_injection_indices(
                    m;
                    node=n,
                    stochastic_scenario=s,
                    t=t_after,
                    temporal_block=anything,
                );
                init=0,
            )
            + expr_sum(
                fractional_demand[(node=n, stochastic_scenario=s, analysis_time=t0, t=t)]
                * demand[(node=ng, stochastic_scenario=s, analysis_time=t0, t=t)]
                for (n, s, t) in node_injection_indices(
                    m;
                    node=n,
                    stochastic_scenario=s,
                    t=t_after,
                    temporal_block=anything,
                ) for ng in groups(n);
                init=0,
            )
```
The individual demand should be working properly also for ordered representative days. The fractional demand however dow not use the representative times, hence this could lead to infeasibilities. Could be easily fixed by simply replacing `t` in the demand and fractional demand call with `representative_time_slice(m, t)`

The second comment I have relates to the diffusion and I was hoping maybe @Topi could have a glance at this:
```
            # Diffusion of commodity from other nodes to this one
            + expr_sum(
                get(node_state, (other_node, s, t_after), 0)
                * diff_coeff[(node1=other_node, node2=n, stochastic_scenario=s, analysis_time=t0, t=t_after)]
                for other_node in node__node(node2=n) for s in s;
                init=0,
            )
```
I believe for the diffusion between two nodes, we implicitly assume that they both follow the same temporal strucutre, but this does not necessarily always hold. I think instead of summing only over the stochastic path, we need to sum over the `node_state_indices` together with `t=t_overlaps_t(t_after)`. I guess we should also be cautious about the differences in TimeSlice durations (should we scale the diffusion accordingly?).

**Comment by clizbe** [comment]
@datejada Can you take a look at this one so we can prioritize it?
